src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -0,0 +1,490 @@
+
+                        AnalyzeUnusedParameters(context, isComputingUnusedParams, symbolUsageResultsBuilder, hasBlockWithAllUsedWrites);
+                    }
+                    finally
+                    {
+                        symbolUsageResultsBuilder.Free();
+                    }
+                }
+
+                private void AnalyzeUnusedValueAssignments(
+                    OperationBlockAnalysisContext context,
+                    bool isComputingUnusedParams,
+                    PooledHashSet<SymbolUsageResult> symbolUsageResultsBuilder,
+                    out bool hasBlockWithAllUsedSymbolWrites)
+                {
+                    hasBlockWithAllUsedSymbolWrites = false;
+
+                    foreach (var operationBlock in context.OperationBlocks)
+                    {
+                        if (!ShouldAnalyze(operationBlock, context.OwningSymbol))
+                        {
+                            continue;
+                        }
+
+                        // First perform the fast, aggressive, imprecise operation-tree based analysis.
+                        // This analysis might flag some "used" symbol writes as "unused", but will not miss reporting any truly unused symbol writes.
+                        // This initial pass helps us reduce the number of methods for which we perform the slower second pass.
+                        // We perform the first fast pass only if there are no delegate creations/lambda methods.
+                        // This is due to the fact that tracking which local/parameter points to which delegate creation target
+                        // at any given program point needs needs flow analysis (second pass).
+                        if (!_hasDelegateCreationOrAnonymousFunction)
+                        {
+                            var resultFromOperationBlockAnalysis = SymbolUsageAnalysis.Run(operationBlock, context.OwningSymbol, context.CancellationToken);
+                            if (!resultFromOperationBlockAnalysis.HasUnreadSymbolWrites())
+                            {
+                                // Assert that even slow pass (dataflow analysis) would have yielded no unused symbol writes.
+                                Debug.Assert(!SymbolUsageAnalysis.Run(context.GetControlFlowGraph(operationBlock), context.OwningSymbol, context.CancellationToken)
+                                             .HasUnreadSymbolWrites());
+
+                                hasBlockWithAllUsedSymbolWrites = true;
+                                continue;
+                            }
+                        }
+
+                        // Now perform the slower, precise, CFG based dataflow analysis to identify the actual unused symbol writes.
+                        var cfg = context.GetControlFlowGraph(operationBlock);
```

Comment:
nit.  we're generally against abbreviations.  controlFlowGraph, flowGraph, contralGraph would be preferred.